### PR TITLE
Use strcpy instead of sprintf %s

### PR DESCRIPTION
**Pull request by richsalz**
Also use a local variable, collapse some code.

### CODE REVIEW

File: crypto/engine/eng_ctrl.c [hunk 1]
```diff
@@ -91,35 +93,29 @@ static int int_ctrl_helper(ENGINE *e, int cmd, long i, void *p,
...
+        return strlen(cdp->cmd_name);
     case ENGINE_CTRL_GET_NAME_FROM_CMD:
-        return sprintf(s, "%s", e->cmd_defns[idx].cmd_name);
+        return strlen(strcpy(s, cdp->cmd_name));
     case ENGINE_CTRL_GET_DESC_LEN_FROM_CMD:
-        if (e->cmd_defns[idx].cmd_desc)
-            return strlen(e->cmd_defns[idx].cmd_desc);
-        return strlen(int_no_description);
+        return strlen(cdp->cmd_desc == NULL
+                ? int_no_description : cdp->cmd_desc);
```
Comment (levitte) [inline crypto/engine/eng_ctrl.c]: Nit pick: sloppy indent

File: crypto/engine/eng_ctrl.c [hunk 2]
```diff
@@ -91,35 +93,29 @@ static int int_ctrl_helper(ENGINE *e, int cmd, long i, void *p,
...
     case ENGINE_CTRL_GET_DESC_FROM_CMD:
-        if (e->cmd_defns[idx].cmd_desc)
-            return sprintf(s, "%s", e->cmd_defns[idx].cmd_desc);
-        return sprintf(s, "%s", int_no_description);
+        return strlen(strcpy(s, cdp->cmd_desc == NULL
+                                    ? int_no_description : cdp->cmd_desc));
```
Comment (levitte) [inline crypto/engine/eng_ctrl.c]: Nit pick: sloppy indent

Comment (levitte) [reply crypto/engine/eng_ctrl.c]: (at the very least, you could have the same indentation standard for both `?:` ops ;-) )

**Comment by richsalz** [comment]
Does this look better?

**Comment by richsalz** [comment]
they builds agreed :)

**Pull request closed without merging**